ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -373,6 +383,54 @@ void fillChunks() {
   }
 
   /**
+   * Submits encrypt job for the given {@link PutChunk} and processes the callback for the same
+   * @param putChunk the {@link PutChunk} that needs to be encrypted
+   * @param encryptContent {@code true} if content needs to be encrypted. {@code false} otherwise
+   */
+  private void encryptChunk(PutChunk putChunk, boolean encryptContent) {
+    try {
+      logger.trace("Chunk at index {} moves to {} state", putChunk.chunkIndex, ChunkState.Encrypting);
+      putChunk.state = ChunkState.Encrypting;
```

Comment:
did we evaluate if this state is necessary?

Reply:
Looks like it is of no use for now. Will remove it.

Reply:
Sorry. I had to make some more changes and it is not just removing this line. At places, we check just for state.Building()(with encryption in picture, since we move the state to Encrypt, those are taken care.) But if I have to remove the state.Encrypting, those conditions needs fix. Not sure if we need to touch those pieces for now. 

to answer your question, during our last offline meeting, most of our discussion was revolving around GetBlobOperation and not really about states in PutOperation.

Reply:
If it is not needed, we should remove IMO